### PR TITLE
Contrib fix linker warnings

### PR DESCRIPTION
**Pull request by clucock1**
Fixed LNK 4221 warnings.
Fixed warnings in icam, volm, and sdet

### CODE REVIEW

File: contrib/brl/bbas/volm/exe/volm_refine_height_map.cxx [hunk 1]
```diff
@@ -389,7 +389,8 @@ int main(int argc, char** argv)
   std::cout << "\nStart to refine ground elevation";
   for (unsigned w_idx_i = 0; w_idx_i < num_w_i; w_idx_i++)
   {
-    unsigned start_ni, end_ni;
+    // warning C4101: 'end_ni': unreferenced local variable. 'start_ni': unreferenced local variable.
+    // unsigned start_ni, end_ni;
```
Comment (decrispell) [inline contrib/brl/bbas/volm/exe/volm_refine_height_map.cxx]: no need to keep the unused variable declarations as comments

File: contrib/brl/bbas/volm/pro/processes/volm_refine_bvxm_height_map_process.cxx [hunk 2]
```diff
@@ -271,7 +271,8 @@ bool volm_extract_building_outlines_process(bprb_func_process& pro)
     double area = 0.0;
     for (auto & k : region) {
       for (unsigned i = k.ilo; i < k.ihi; i++) {
-        double lon, lat;
+        // warning C4101: 'lon': unreferenced local variable. 'lat': unreferenced local variable.
+        // double lon, lat;
```
Comment (decrispell) [inline contrib/brl/bbas/volm/pro/processes/volm_refine_bvxm_height_map_process.cxx]: no need to keep as comments

File: contrib/brl/bbas/volm/volm_geo_index.cxx [hunk 3]
```diff
@@ -453,7 +453,7 @@ bool volm_geo_index::exists(const volm_geo_index_node_sptr& root, double lat, do
   if (!root)
     return false;
 
-  vgl_point_3d<double> h;
+//: warning C4101: 'h': unreferenced local variable.  vgl_point_3d<double> h;
```
Comment (decrispell) [inline contrib/brl/bbas/volm/volm_geo_index.cxx]: no need to keep as comment

File: contrib/brl/bseg/sdet/sdet_atmospheric_image_classifier.cxx [hunk 4]
```diff
@@ -320,7 +320,7 @@ classify_image_blocks_expected(vil_image_view<float> const& image,
       float b_good = pr - 0.5f*u;
       if (b_good< 0.0f) b_good = 0.0f;
       float b_bad = 1.0f - u - b_good;
-      vnl_vector_fixed<float, 3> color;
+      // vnl_vector_fixed<float, 3> color; warning C4101: 'color': unreferenced local variable.
```
Comment (decrispell) [inline contrib/brl/bseg/sdet/sdet_atmospheric_image_classifier.cxx]: no need to keep as comment

File: contrib/brl/bbas/volm/pro/processes/volm_refine_bvxm_height_map_process.cxx (hunk 2 above)
Comment (clucock1) [reply contrib/brl/bbas/volm/pro/processes/volm_refine_bvxm_height_map_process.cxx]: I will delete the comments.

**Pull request merged**